### PR TITLE
[fix #4819] update react-native to 0.56

### LINKED ISSUE #4819: Update react-native to 0.56
Update react-native version to 0.56

With recent upgrade to `0.55` RN release, we introduced multiple problems, one is perf regression related to chat input element (#4794) and it's probably causing keyboard crash issue (#4818) as well.
We hope that upgrading to newest RN release will solve those things.

To check:
- #4794 is no more
- no regressions
- no new perf regressions

Bounty might be increased if it turns out that the update required more changes than expected.

### PR DESCRIPTION
**Pull request by yenda**
fixes #4819 

## Summary

update status-react mobile version to react-native 0.56

## Step for testing

verify that the two bugs introduced in previous update are gone:
- perf regression related to chat input element (#4794)
- keyboard crash issue (#4818)
- make sure that Korean input on iOS works correctly

status: ready

### CODE REVIEW

File: mobile_files/package.json
Comment (dmitryn) [inline mobile_files/package.json]: Are we switching to yenda/react-native fork?

**Review by dmitryn — APPROVED** [review]

File: mobile_files/package.json
Comment (yenda) [reply mobile_files/package.json]: no I didn't had the rights to create the branch on status-im fork, but will do before rebasing. It doesn't matter for testing

**Review by mandrigin — CHANGES_REQUESTED** [review]
someone needs to make sure that Korean input on iOS works correctly there.
see this PR for details https://github.com/facebook/react-native/pull/19809

**Comment by vkjr** [comment]
@churik, could you please check that desktop build in this PR works as expected? Thanks.

**Comment by yenda** [comment]
@mandrigin it uses a version of 0.56 that includes our fix to this issue, please remove the request for changes so that testing can test it

**Comment by churik** [comment]
Tested [build for desktop](https://ci.status.im/job/status-react/job/desktop/job/pull-requests/view/change-requests/job/PR-5610/) - it can be started, messages are fetched.

**Review by mandrigin — APPROVED** [review]

**Comment by jakubgs** [comment]
Hope this change gets merged soon, I didn't know it was already happening so I mistakenly created this:
https://github.com/status-im/status-react/issues/5636
Closed it as a duplicate.
Once this is merged we will be able - hopeuflly - to build on Linux, which means only one of the 5 parallel builds we run will be on MacOS, which should reduce how often we had far too long build queue since we can sacale up Linux Jenkins slaves far easier.

**Comment by lukaszfryc** [comment]
Jenkins still fails on 

```
:react-native-config:verifyReleaseResources/home/jenkins/.gradle/caches/transforms-1/files-1.1/appcompat-v7-26.1.0.aar/980076bad91acd70a015ae788adb7fff/res/values-v24/values-v24.xml:3:5-157: AAPT: Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.Button.Borderless.Colored'.
    
/home/jenkins/.gradle/caches/transforms-1/files-1.1/appcompat-v7-26.1.0.aar/980076bad91acd70a015ae788adb7fff/res/values-v24/values-v24.xml:4:5-135: AAPT: Error retrieving parent for item: No resource found that matches the given name 'android:TextAppearance.Material.Widget.Button.Colored'.
```

@dmitryn could you check it and provide me all the builds (mobile, desktop) to test?

**Comment by dmitryn** [comment]
@lukaszfryc yes, android build fails on different dependencies. First it was `react-native-backdround-timer`, now it's `react-native-config`. We're working on making the build.

**Review by dmitryn — CHANGES_REQUESTED** [review]
Move forked versions under `status-im` org account

**Review by dmitryn — APPROVED** [review]

**Comment by dmitryn** [comment]
All builds are green except `e2e`, error with `fastlane`:

```
bundle exec fastlane android saucelabs
bundler: failed to load command: fastlane (/usr/local/bin/fastlane)
Bundler::GemNotFound: Could not find atomos-0.1.2 in any of the sources
  /usr/lib/ruby/vendor_ruby/bundler/spec_set.rb:88:in `block in materialize'
  /usr/lib/ruby/vendor_ruby/bundler/spec_set.rb:82:in `map!'
  /usr/lib/ruby/vendor_ruby/bundler/spec_set.rb:82:in `materialize'
  /usr/lib/ruby/vendor_ruby/bundler/definition.rb:170:in `specs'
  /usr/lib/ruby/vendor_ruby/bundler/definition.rb:237:in `specs_for'
  /usr/lib/ruby/vendor_ruby/bundler/definition.rb:226:in `requested_specs'
  /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:108:in `block in definition_method'
  /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:20:in `setup'
  /usr/lib/ruby/vendor_ruby/bundler.rb:107:in `setup'
  /usr/lib/ruby/vendor_ruby/bundler/setup.rb:20:in `<top (required)>'
  /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
  /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
```

**Comment by yenda** [comment]
@dmitryn @jakubgs maybe fastlane isn't installed on that host yet ?

**Comment by dmitryn** [comment]
@yenda it seems like it's there `/usr/local/bin/fastlane`, but hard to say without access to that host.

Locally i don't have this issue, but i'm using mac os.

**Comment by dmitryn** [comment]
Similar job from another PR was successful https://ci.status.im/job/status-react/job/combined/job/mobile-android/1017/console 

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/combined/job/mobile-android/1017/console">status-react » combined » mobile-android #1017 Console [Jenkins]</a></strong></div></blockquote>

**Comment by jakubgs** [comment]
I think this looks like something I can help with.

**Comment by jakubgs** [comment]
Fixed the bundle error with:
https://github.com/status-im/status-react/commit/3d6c2855be55cc0881c4bd28d4c0738a672a89d9
So I took the liberty of rebasing your branch against `develop`.

**Comment by jakubgs** [comment]
Why did you undo changes in `Jenkinsfile.linux` in the environment section? You break the build if variables like `ANDROID_SDK` or `ANDROID_NDK` are not set correctly.
The correct values are `/usr/lib/android-sdk` and `/usr/lib/android-ndk`.

**Comment by dmitryn** [comment]
@jakubgs it was probably done because of wrong rebase. Reverting back..

**Comment by asemiankevich** [comment]
@yenda 2 major issues found:

1. no qr code shown for iOS device (open status - wallet - receive transaction)

image
Update: QR Code is also not shown in "Share my contact code"
nocode

2. any transaction (send/request) from wallet / chat that is performed TO android device triggers error
image

Please take a look into those 2 while i continue testing this PR.

Update: any message when receiver is Android device leads to same error

**Comment by churik** [comment]
3. At attempt to paste empty string in webview got "null is not an object (evaluating 'e.length')

Scenario 1:
- navigate to chat
- open any link in Status
- erase value and paste several whitespaces 
- tap on go

Scenario 2:
- Tap on +
- Tap on open Dapp
- in "Enter URL field" paste whitespaces
- tap on Go

Reproducible for both IOS and Android.
upl
TF session: 
https://app.testfairy.com/projects/4803590-status/builds/8579833/sessions/4400474291/?accessToken=8gco9fUSEv9tTNwvs3YoZz6Dlqw

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803590-status/builds/8579833/sessions/4400474291/?accessToken=8gco9fUSEv9tTNwvs3YoZz6Dlqw">TestFairy: Status 0.9.26 (7624) @2018</a></strong></div></blockquote>

**Comment by dmitryn** [comment]
All issues have been fixed. Waiting for the build to arrive.

**Comment by dmitryn** [comment]
@asemiankevich @churik new build is ready, please take a look

**Comment by asemiankevich** [comment]
@dmitryn thank you! I can confirm the following issues are fixed:
1. QR code on iOS is now shown
2. No error when transactioning between ios - android
3. Spaces in URL does not throw error

However I noticed that there are no pictures for collectibles on iOS (android looks fine). Develop looks good. Could you please take a look?

image

**Comment by dmitryn** [comment]
@asemiankevich collectibles error has been fixed, please take a look.

**Comment by churik** [comment]
Collectibles error, https://github.com/status-im/status-react/issues/4794 and https://github.com/status-im/status-react/issues/4818 , issue with Korean keyboard (verified according to [test case](https://ethstatus.testrail.net/index.php?/cases/view/3752&group_by=cases:section_id&group_order=asc&group_id=308) ) are not reproducible.

**Comment by asemiankevich** [comment]
e2e fails are false. I did not find any new issues in this PR. Performance tests look good. I think we can merge this one to develop.

**Comment by dmitryn** [comment]
Making new build after fixing conflicts and rebasing. Will merge after that.

**Pull request merged**
